### DISCUSSION TITLE
wiki/pyqgis-sd/04_prev/canvas_layers

### DISCUSSION THREAD

**Discussion by giscus** [SX-Code/blog-comments#13] [Announcements]
# wiki/pyqgis-sd/04_prev/canvas_layers

QGIS为我们提供了默认的图层管理器，我们需要绑定图层树和画布，然后向图层树中添加图层，画布和图层视图都会显示添加的数据。 为此，先删除之前设计界面时添加的 List View，找到frame_prev_layers删除其中的list_view，并修改其layoutName为layout_prev_layers，稍后会向该布局中添加图层视图。 修改init_qgis_map方法，添加如下代码 de

https://sx-code.github.io/wiki/pyqgis-sd/04_prev/canvas_layers.html

**Comment by FeihuXue-dev** [comment]
请问这段代码是添加在mainwin中吗？

**Reply by SX-Code** [reply]
这个项目说明文档是有上下文的，需要联系上下文。有源码可下载，方便理解

**Reply by FeihuXue-dev** [reply]
谢谢，找到了，不好意思

**Comment by FeihuXue-dev** [comment]
import gui as GUI 请问导入后，运行并没有实现preview中图层树和画布的绑定，如何修改

**Reply by SX-Code** [reply]
确保使用QgsLayerTreeMapCanvasBridge建立图层树与地图画布的桥接，在向canvas添加layer的同时，需要调用QgsProject.instance().addMapLayer(layer)，向全局实例中添加。
